core/domain/stats_services_test.py

```diff
@@ -1497,6 +1514,35 @@ def setUp(self):
         self.save_new_linear_exp_with_state_names_and_interactions(
             self.EXP_ID, self.owner_id, self.STATE_NAMES, ['TextInput'])
 
+    def test_get_top_state_unresolved_answer_stats(self):
+        self._record_answer(
+            'A', classification_category=exp_domain.EXPLICIT_CLASSIFICATION)
+        self._record_answer(
+            'B', classification_category=exp_domain.EXPLICIT_CLASSIFICATION)
+        self._record_answer(
+            'C', classification_category=exp_domain.STATISTICAL_CLASSIFICATION)
+        self._record_answer(
+            'A', classification_category=exp_domain.STATISTICAL_CLASSIFICATION)
+        self._record_answer(
+            'D',
+            classification_category=exp_domain.DEFAULT_OUTCOME_CLASSIFICATION)
+        self._record_answer(
+            'E',
+            classification_category=exp_domain.DEFAULT_OUTCOME_CLASSIFICATION)
+        self._record_answer(
+            'D', classification_category=exp_domain.EXPLICIT_CLASSIFICATION)
+        self._run_answer_summaries_aggregator()
+
+        with self.swap(feconf, 'STATE_ANSWER_STATS_MIN_FREQUENCY', 1):
+            state_answers_stats = self._get_top_state_unresolved_answer_stats()
```

Comment:
Hm -- there should also be some tests that check that subsequently resolving answers leads to the correct results, right?

Reply:
Done.